homeassistant/components/shelly/sensor.py

```diff
@@ -404,12 +485,12 @@ def native_value(self) -> StateType:
 
         return self.last_state
 
-    @property
-    def state_class(self) -> str | None:
-        """State class of sensor."""
-        return self.description.state_class
+    async def async_added_to_hass(self) -> None:
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
 
-    @property
-    def native_unit_of_measurement(self) -> str | None:
-        """Return unit of sensor."""
-        return cast(str, self._unit)
+        last_state = await self.async_get_last_state()
+
+        if last_state is not None:
+            self.last_state = last_state.state
+            self._attr_state_class = last_state.attributes.get(ATTR_STATE_CLASS)
```

Comment:
I wonder if we need to manually restore the state class, I think this was a temporary workaround when we added it, maybe we can remove it and move this part back to `ShellySleepingBlockAttributeEntity`

Reply:
Done in b381ee6